src/axom/sidre/tests/sidre_group.cpp

```diff
@@ -1157,6 +1158,140 @@ TEST(sidre_group,save_load_via_hdf5_ids)
   // close hdf5 handle
   EXPECT_TRUE(H5Fclose(h5_id) >=0);
 }
+
+//------------------------------------------------------------------------------
+TEST(sidre_group,save_root_restore_as_child)
+{
+  // We'll save the DataStore's root Group into a file then restore it
+  // into a child group of another DataStore.
+
+  // Create a DataStore; put some groups and views into it
+  const std::string file_path_base("sidre_save_root_restore_as_child_");
+  DataStore* ds = new DataStore();
+  Group* root = ds->getRoot();
+  Group* child1 = root->createGroup("g_a");
+  Group* child2 = root->createGroup("g_b");
+  root->createGroup("g_c"); // We don't put anything into g_c
+
+  child1->createViewScalar<int>("i0", 1);
+  child1->createViewString("s0", "I am a string");
+
+  const int num_elems = 4;
+  int * pa0 =
+    child2->createViewAndAllocate("a0", INT_ID, num_elems)->getArray();
+  double * pa1 =
+    child2->createViewAndAllocate("a1", FLOAT64_ID, num_elems)->getArray();
+
+  const double factor = 2.3;
+  const double offset = -0.23;
+  for (int i = 0; i < num_elems; ++i)
+  {
+     pa0[i] = i;
+     pa1[i] = offset + i*factor;
+  }
+
+  // Save the DataStore's root
+  for (int i = 0 ; i < nprotocols ; ++i)
+  {
+    const std::string file_path = file_path_base + protocols[i];
+    root->save(file_path, protocols[i]);
+  }
+
+  // Restore the original DataStore into a child group
+  for (int i = 0; i < nprotocols; ++i)
+  {
+    // Only restore sidre_hdf5 protocol
```

Comment:
Looks like this check can be removed once #97 is addressed.

Reply:
yes